<!-- -*- mode: XML -*- -->
<launch>
	<arg name="launcher" 	default="ardrone2_launch" />
	<!-- <node ns="ardrone2" pkg="tf" type="static_transform_publisher" name="tf_map2odom" args="0 0 0 0 0 0 /map /ardrone_autonomy/odom 100"/>  -->
	

	<node pkg="ardrone_autonomy" type="ardrone_driver" name="ardrone_autonomy" clear_params="true">
		<param name="tf_prefix" value="ardrone_autonomy" />  
		<param name="realtime_navdata" value="true" />
		<param name="realtime_video" value="true" />		
		<param name="looprate" value="50" /> <!-- Tunes the speed at which the ros loop runs, and thus, the rate at which navdata is published -->
	    <remap from="/cmd_vel" to="hast/uav/cmd_vel"/>
	</node>

	<node pkg="rostopic" type="rostopic" 					name="navdataEcho" 											output="screen" args="echo /ardrone/navdata" />
	<node pkg="hast"		 type="killhast"					name="killhast_$(arg launcher)"					output="screen" required="true"/>
	<node pkg="hast"		 type="shutdownListener"	name="shutdownListener_$(arg launcher)"	output="screen"	required="true"/>

	<!-- UAV TF for Rviz  -->
	<arg name="UAV_ns" 						default="UAV" />
	<arg name="uav_model" 				default="$(find robot_descriptions)/UAV/urdf/ardrone_rviz.urdf.xacro"/> 
	<param name="UAV_description" command="$(find xacro)/xacro --inorder $(arg uav_model)" />

	<node ns="$(arg UAV_ns)" pkg="tf" type="static_transform_publisher" name="$(arg UAV_ns)_driver2odom" 
		args="0.0 0.0 0.0 0.0 0.0 0.0 /map /ardrone_autonomy/odom 1"/> 
	<node ns="$(arg UAV_ns)" pkg="tf" type="static_transform_publisher" name="$(arg UAV_ns)_driver2base" 
		args="0.0 0.0 0.0 0.0 0.0 0.0 /hast/uav/ardrone_base_link /vicon/uav/base_footprint 1"/> 

	<node ns="$(arg UAV_ns)" pkg="tf" type="static_transform_publisher" name="$(arg UAV_ns)_base2red" 
		args="0.0 -0.22 -0.01 0.0 0.0 0.0 /vicon/uav/base_footprint /vicon/uav/ardrone_red_led 1"/> 
	<node ns="$(arg UAV_ns)" pkg="tf" type="static_transform_publisher" name="$(arg UAV_ns)_base2blue" 
		args="0.0 0.22 -0.01 0.0 0.0 0.0 /vicon/uav/base_footprint /vicon/uav/ardrone_blue_led 1"/> 
	<node ns="$(arg UAV_ns)" pkg="tf" type="static_transform_publisher" name="$(arg UAV_ns)_base2green" 
		args="-0.22 0.0 -0.01 0.0 0.0 0.0 /vicon/uav/base_footprint /vicon/uav/ardrone_green_led 1"/> 

	<!-- RVIZ for april tags -->
	<include file="$(find hast_gazebo)/launch/rviz_tags.launch"/>

</launch>

	<!-- <node pkg="rosservice" type="rosservice" name="toggle" args="call /ardrone/togglecam"/> 		 -->


<!-- node pkg="tf" type="static_transform_publisher" name="world_to_cam"  args="x y z yaw pitch roll frame_id child_frame_id period_in_milisec" /-->
 
 <!--  Is this a half meter??? args="0 0 0.5 -1.57 0 -1.57 world camera 1" /> -->
	<!-- <node pkg="ardrone_autonomy" type="ardrone_driver" name="ardrone_autonomy" output="screen" clear_params="true"> -->
        <!-- <param name="do_imu_caliberation" value="true" /> -->
        <!-- <rosparam param="cov/imu_la">[0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 0.0, 0.0, 0.1]</rosparam> -->
        <!-- <rosparam param="cov/imu_av">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam> -->
        <!-- <rosparam param="cov/imu_or">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 100000.0]</rosparam> -->
	    <!-- <param name="enable_navdata_pwm" value="true"/> -->
		<!-- <rosparam file="/home/turtlebot/ros/src/hast/cam_info/00b09d0100c8851b.yaml" />  -->
